examples/relay-chat/index.js

```diff
@@ -0,0 +1,340 @@
+  });
+
+  node.libp2p.addEventListener("peer:identify", (event) => {
+    const peer = event.detail;
+
+    if (!peer.protocols.includes("/webrtc-signaling/0.0.1")) {
+      return;
+    }
+
+    ui.setWebrtcPeer(peer.peerId.toString());
+    ui.setRelayMeshInfo(node.relay.gossipSub);
+  });
+
+  window.node = node;
+
+  return {
+    unsubscribeFromMessages,
+    info: {
+      contentTopic,
+      localPeerId,
+    },
+    sendMessage: async ({ text, nick }) => {
+      if (!text || !nick) {
+        return;
+      }
+
+      const protoMessage = ChatMessage.create({
+        nick,
+        timestamp: Date.now(),
+        text: utf8ToBytes(text),
+      });
+
+      await node.relay.send(Encoder, {
+        payload: ChatMessage.encode(protoMessage).finish(),
+      });
+    },
+    dial: async (multiaddr) => {
+      ui.setStatus("connecting...", "progress");
+      await node.dial(multiaddr);
+      ui.setStatus("connected", "success");
+    },
+    dialWebRTCpeer: async (peerId) => {
+      const peers = await node.libp2p.peerStore.all();
+      const circuitPeer = peers.filter(
+        (p) =>
+          p.protocols.includes("/libp2p/circuit/relay/0.2.0/hop") &&
+          p.protocols.includes("/libp2p/circuit/relay/0.2.0/stop")
+      )[0];
+
+      if (!circuitPeer) {
+        throw Error("No Circuit peer is found");
+      }
+
+      let multiaddr = circuitPeer.addresses.pop().multiaddr;
+      multiaddr = `${multiaddr}/p2p/${circuitPeer.id.toString()}/p2p-circuit/webrtc/p2p/${peerId}`;
+
+      await node.dial(multiaddr);
+      ui.setRelayMeshInfo(node.relay.gossipSub);
+    },
+    ensureWebRTCconnectionInRelayMesh: async () => {
+      const promises = node.libp2p
+        .getConnections()
+        .filter((c) => c.stat.multiplexer === "/webrtc")
+        .map(async (c) => {
+          const outboundStream = node.relay.gossipSub.streamsOutbound.get(
+            c.remotePeer.toString()
+          );
+          const isWebRTCOutbound =
+            outboundStream.rawStream.constructor.name === "WebRTCStream";
+
+          if (isWebRTCOutbound) {
+            return;
+          }
+
+          node.relay.gossipSub.streamsOutbound.delete(c.remotePeer.toString());
+          await node.relay.gossipSub.createOutboundStream(
+            c.remotePeer.toString(),
+            c
+          );
+        });
+      await Promise.all(promises);
+      ui.setRelayMeshInfo(node.relay.gossipSub);
+    },
+    dropNetworkConnections: async () => {
+      const promises = node.libp2p
+        .getConnections()
+        .filter((c) => c.stat.multiplexer !== "/webrtc")
+        .map(async (c) => {
+          const peerId = c.remotePeer.toString();
+
+          node.relay.gossipSub.peers.delete(peerId);
+          node.relay.gossipSub.streamsInbound.delete(peerId);
+          node.relay.gossipSub.streamsOutbound.delete(peerId);
+
+          await node.libp2p.peerStore.delete(c.remotePeer);
+          await c.close();
```

Comment:
it's good enough for presentation purpose to close connection and clean gossipSub's mesh

Reply:
gossipsub mesh should clean itself at next heartbeat

Reply:
will check, didn't notice it can be the case

Reply:
leaving it as it is now as it seems that sometimes connection is re-created and added again to the mesh